sdk/nodejs/cmd/pulumi-language-nodejs/main_test.go

```diff
@@ -369,6 +369,296 @@ func TestGetRequiredPluginsNestedPolicyPack(t *testing.T) {
+		t.Parallel()
+
+		files := []filePathAndContents{
+			{
+				path:    "package.json",
+				content: `{ "name": "@pulumi/baz", "dependencies": { "@pulumi/pulumi": "^3.113.0" } }`,
+			},
+			{
+				path:    "Pulumi.yaml",
+				content: `name: test`,
+			},
+		}
+
+		testDir := setupFiles(t, files)
+		host := &nodeLanguageHost{}
+		_, err := host.GetProgramDependencies(context.Background(), &pulumirpc.GetProgramDependenciesRequest{
+			Program: testDir,
+			Info: &pulumirpc.ProgramInfo{
+				RootDirectory:    testDir,
+				ProgramDirectory: testDir,
+				EntryPoint:       ".",
+			},
+		})
+		require.ErrorContains(t, err, "no package-lock.json or yarn.lock file found (searching upwards from")
+	})
+
+	t.Run("package.json-in-project-root-with-yarn.lock", func(t *testing.T) {
+		t.Parallel()
+
+		files := []filePathAndContents{
+			{
+				path:    "package.json",
+				content: `{ "name": "@pulumi/baz", "dependencies": { "@pulumi/pulumi": "^3.113.0" } }`,
+			},
+			{
+				path:    "Pulumi.yaml",
+				content: `name: test`,
+			},
+			{
+				path: "yarn.lock",
+				content: `"@pulumi/pulumi@^3.0.0", "@pulumi/pulumi@^3.113.0":
+  version "3.131.0"
+  resolved "https://registry.yarnpkg.com/@pulumi/pulumi/-/pulumi-3.131.0.tgz#6233e5ee5e72907b99415b32be6a9ebf9041f096"
+  integrity sha512-QNtQeav3dkU0mRdMe2TVvkBmIGkBevVvbD7/bt0fJlGoX/onzv5tysqi1GWCkXsq0FKtBtGYNpVD6wH0cqMN6g==
+  dependencies:
+    "@grpc/grpc-js" "^1.10.1"
+    "@logdna/tail-file" "^2.0.6"
+    "@npmcli/arborist" "^7.3.1"`,
+			},
+		}
+
+		testDir := setupFiles(t, files)
+		host := &nodeLanguageHost{}
+		resp, err := host.GetProgramDependencies(context.Background(), &pulumirpc.GetProgramDependenciesRequest{
+			Program: testDir,
+			Info: &pulumirpc.ProgramInfo{
+				RootDirectory:    testDir,
+				ProgramDirectory: testDir,
+				EntryPoint:       ".",
+			},
+		})
+		require.NoError(t, err)
+		require.Equal(t, len(resp.Dependencies), 1)
+		require.Equal(t, resp.Dependencies[0].Name, "@pulumi/pulumi")
+		require.Equal(t, resp.Dependencies[0].Version, "3.131.0")
+	})
+
+	t.Run("package.json-in-parent-with-yarn.lock", func(t *testing.T) {
+		t.Parallel()
+
+		files := []filePathAndContents{
+			{
+				path:    "package.json",
+				content: `{ "name": "@pulumi/baz", "dependencies": { "@pulumi/pulumi": "^3.113.0" } }`,
+			},
+			{
+				path:    filepath.Join("subdir", "Pulumi.yaml"),
+				content: `name: test`,
+			},
+			{
+				path: "yarn.lock",
+				content: `"@pulumi/pulumi@^3.0.0", "@pulumi/pulumi@^3.113.0":
+  version "3.131.0"
+  resolved "https://registry.yarnpkg.com/@pulumi/pulumi/-/pulumi-3.131.0.tgz#6233e5ee5e72907b99415b32be6a9ebf9041f096"
+  integrity sha512-QNtQeav3dkU0mRdMe2TVvkBmIGkBevVvbD7/bt0fJlGoX/onzv5tysqi1GWCkXsq0FKtBtGYNpVD6wH0cqMN6g==
+  dependencies:
+    "@grpc/grpc-js" "^1.10.1"
+    "@logdna/tail-file" "^2.0.6"
+    "@npmcli/arborist" "^7.3.1"`,
+			},
+		}
+
+		testDir := setupFiles(t, files)
+		host := &nodeLanguageHost{}
```

Comment:
Maybe define `subdir := filepath.Join(testDir, "subdir")` here to make it obvious how the test is going to proceed and use it below?